Basic-Car-Maintenance/Shared/MainView/ViewModels/MainTabViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MainTabViewModel.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Omar Hegazy on 13/10/2023.
+//
+
+import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import SwiftData
+
+@Observable
+class MainTabViewModel {
+    private var acknowledgedAlerts = [AcknowledgedAlert]()
+    @MainActor var alert: AlertItem?
+    let modelContext: ModelContext
+    
+    init(modelContext: ModelContext) {
+        self.modelContext = modelContext
+        fetchAcknowledgedAlerts()
+    }
+    
+    /// Update the UI once a new alert is sent
+    func listenToAlertsUpdates() {
+        var query = Firestore
+            .firestore()
+            .collection("alerts")
+            .whereField("isOn", isEqualTo: true)
```

Comment:
Use the new `FirestoreCollection` and `FirestoreField` enums

You'll need to create new properties